### ISSUE TITLE
Source maps not working

### ISSUE THREAD

**Issue by nite** [thlorenz/es6ify#44]
Hi, 

Source maps dont appear to be working with the current browserify - the compiled versions appear in source (in latest chrome) - any ideas?

**Comment by thlorenz** [comment]
Could you please push up a repro case on github?
This very much sounds like a browserify issue, so once you push it up and we don't see anything obvious we should file an issue there as well.

**Comment by thlorenz** [comment]
Make the repro case so I can just run `npm install && npm test` to see it launch and see the problem.
I'm assuming you are browserifying with the `--debug` flag right?

**Comment by nite** [comment]
no changes - all I do to repro is git pull, npm install, run node example/build.js then open index.html & dev tools

**Comment by nite** [comment]
Ahh I hadnt run npm test - without changes again, it returns this: 

> es6ify@1.3.0 test C:\Dev\QB\es6ify
>     > tap test/*.js

```
ok test/bundle.js ..................................... 17/17
ok test/errors.js ....................................... 3/3
not ok test/transform.js ................................ 2/4
    Command: "C:\Program Files\nodejs\node.exe transform.js"
    TAP version 13
    not ok 1 adds sourcemap comment including original source
      ---
        file:   stream.js
        line:   79
        column: 10
        stack:
          - getCaller (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-assert.js:418:1
7)
          - assert (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-assert.js:21:16)
          - Function.equivalent (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-asser
t.js:182:12)
          - Test._testAssert (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-test.js:
87:16)
          - Stream.end (C:\Dev\QB\es6ify\test\transform.js:44:9)
          - _end (C:\Dev\QB\es6ify\node_modules\through\index.js:61:9)
          - Stream.stream.end (C:\Dev\QB\es6ify\node_modules\through\index.js:70
:5)
          - Stream.onend (stream.js:79:10)
          - Stream.EventEmitter.emit (events.js:117:20)
          - drain (C:\Dev\QB\es6ify\node_modules\through\index.js:33:23)
        found:
          version:        3
          sources:
            -
          names:
            -
          mappings:
          file:           C:\Dev\QB\es6ify\example\src\features\iterators.js.es6

          sourcesContent:
            -
        wanted:
          version:        3
          file:           C:\Dev\QB\es6ify\example\src\features\iterators.js.es6

          sources:
            - C:\Dev\QB\es6ify\example\src\features\iterators.js
          names:
            -
          mappings:
          sourcesContent:
            - |
              module.exports = function () {
                for (let element of [1, 2, 3]) {
                  console.log('element:', element);
                }
              };
        diff:   |
          {
            "version" : 3,
            "sources" : [

            ],
            "names" : [

            ],
            "mappings" : "",
            "file" : "C:\Dev\QB\es6ify\example\src\features\iterators.js.es6",
            "sourcesContent" : [

            ]
          }
      ...
    not ok 2 (unnamed assert)
      ---
        file:   stream.js
        line:   79
        column: 10
        stack:
          - getCaller (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-assert.js:418:1
7)
          - Function.assert (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-assert.js
:21:16)
          - Test._testAssert [as ok] (C:\Dev\QB\es6ify\node_modules\tap\lib\tap-
test.js:87:16)
          - Stream.end (C:\Dev\QB\es6ify\test\transform.js:54:9)
          - _end (C:\Dev\QB\es6ify\node_modules\through\index.js:61:9)
          - Stream.stream.end (C:\Dev\QB\es6ify\node_modules\through\index.js:70
:5)
          - Stream.onend (stream.js:79:10)
          - Stream.EventEmitter.emit (events.js:117:20)
          - drain (C:\Dev\QB\es6ify\node_modules\through\index.js:33:23)
          - Stream.stream.queue.stream.push (C:\Dev\QB\es6ify\node_modules\throu
gh\index.js:41:5)
      ...
    ok 3 compiles only the first time
    ok 4 test/transform.js

    1..4
    # tests 4
    # pass  2
    # fail  2

total ................................................. 22/24

not ok
npm ERR! Test failed.  See above for more details.
npm ERR! not ok code 0
```

**Comment by nite** [comment]
I'm also not running browserify from the command line - so the --debug flag is covered by the .bundle({ debug: true }) ... unless I'm missing something?

**Comment by thlorenz** [comment]
No, you should be covered.

This looks interesting, looks like the entire `sourcesContent` is missing. I'd guess a regression that was introduced in browserify at some point.

Could you please fork es6ify, upgrade browserify, so people can clone it and run `npm test` to see the problem? We should raise an issue with browserify.

**Comment by nite** [comment]
I've forked to https://github.com/nite/es6ify - it still show the error first time without changes (running npm install, npm test). How should we raise the issue with browserify? Sorry, I'm not an expert in sourcemaps, node or browserify!

**Comment by nite** [comment]
Are you able to replicate Thorsten? Do you know the most recent version of
browserify I can install for it to work in the meantime?

**Comment by thlorenz** [comment]
I don't see any changes in your fork (where did you update browserify?)
You need to get your fork into a state that simply cloning it followed by `npm install && npm test` shows the problem.
Right now it doesn't.

Thanks.

**Comment by thlorenz** [comment]
Unless it fails on your machine as is in which case we got a different problem, since tests pass for me.

**Comment by nite** [comment]
Very strange - it does for me, without changes! I'm on windows, that couldnt be an issue right? I still get the issue if I force 'npm install browserify@3.33.1' (the version in packages.json)

**Comment by thlorenz** [comment]
Ok very strange, does seem like a windows issue at this point.
If you want we could set up [appveyor](http://www.appveyor.com/) to run these tests on windows.

You can get a free account to set this up in your fork and once it's working we'll merge it in here and adjust it to run under my account.
At that point we can point the browserify people at that and possibly someone there can help.

The `.appveyour` file would be very similar to [this one](https://github.com/rvagg/nan/blob/master/appveyor.yml).

**Comment by thlorenz** [comment]
Also please check if you get sourcemaps when not using es6ify (trying to narrow down the problem).

**Comment by nite** [comment]
Right, might have to leave this one til tomorrow - but I've checked and it works fine for me on mac too. I'll try both of these (sourcemaps without es6ify and appveyor).

**Comment by nite** [comment]
I've set up an appveyor - again not an expert, do I put the appveyor yml file into my forked repo?

**Comment by thlorenz** [comment]
yep just do that and then add a hook to your fork in appveyor. That will cause it to build it.

**Comment by nite** [comment]
Queued - waiting ages. I've just opened up index.html on mac and that works too :) this is gonna force me to try dev-ing in mac until I sort this out methinks! Trying to set up a non-es6ify browserify at the moment (again, not an expert - what's the quickest test?)

**Comment by nite** [comment]
Should appveyor take more than 15 minutes? hmm ...

**Comment by nite** [comment]
Source maps work fine in a standalone browserify - I ran through the demo on their front page and stepped into main.js in chrome, also added my own js file & required & stepped into it ...

**Comment by nite** [comment]
fail replicated here: 
https://ci.appveyor.com/project/Andrew/es6ify/build/job/y5boqdayr5ljs8t5

**Comment by nite** [comment]
Thorsten, do you still think its a browserify issue? I'm able to debug into source via maps with a vanilla js test, including a required module.

**Comment by thlorenz** [comment]
I don't think so since it works without es6ify.
I have no way to debug this since I don't own a windows machine (thank God).

The problem could lie with traceur (since that is what actually generates the original source maps).
We just [_fix_](https://github.com/thlorenz/es6ify/blob/master/index.js#L34-L58) them a bit in order to avoid problems where sourcemaps from different files where included with particular files.

So you could debug there in order to see if traceur is already not giving us the correct sourcemaps or if es6ify is doing something wrong there.

Since you said it worked for you before you might try older versions (mostly they use older traceur versions). Just check out any of [these tags](https://github.com/thlorenz/es6ify/tags) and find the latest one that works.
Then checkout master again, but downgrade traceur to the version used in the particular tag.
If changing traceur version makes things break then the problem lies with traceur, otherwise we have to dig into es6ify.

**Comment by nite** [comment]
So I'm assuming it's a problem with traceur, this really is the only thing that could change between mac & win. It never worked for me before on win, only on mac. I'm currently working on mac . Not sure what's different, on my project the sourcemap is out by 1 line - but hey that's better than no es6 in my devtools.

**Comment by thlorenz** [comment]
Interesting, you could try to use traceur directly to confirm, but that is the first time I heard about it.
I thought there were other windows es6ify users, i.e. @domenic ?

**Comment by nite** [comment]
Yeah surely there'd have to be others using this on windows successfully or you'd have heard about it by now, but there's no other logical possibility here really, once node is running it runs es6ify in the same way - the only possibility seems the traceur runtime. I'll give traceur a go on its own when I get a change. In case I'm missing something - where would you debug in index.js and what would you test/look for?

**Comment by thlorenz** [comment]
You'd debug around [here](https://github.com/thlorenz/es6ify/blob/master/index.js#L34-L58) and can just run one of the failing tests.

**Comment by Apoxx** [comment]
Hi, the bug (or not) come from Traceur. Here : https://github.com/google/traceur-compiler/blob/bdae9f987a82795c8d41331b50570575cfb4a5e2/src/node/NodeCompiler.js#L42-L44

It returns "..\.." which is added at the start of the file path. Because of that the paths do not match in the browser.

You can remove the return statement to make it work. I didn't dig further than that so I don't know the side effects, but all es6ify tests pass.

**Comment by urish** [comment]
Got a similar issue here, the fix that worked for me is to change [index.js:42](https://github.com/thlorenz/es6ify/blob/master/index.js#L42) to read:

``` js
    if(mapping.source && (file === mapping.source || (compiled.sourcemap.sourceRoot + '/' + file === mapping.source))) {
```

@thlorenz Should I send a PR?

**Comment by domenic** [comment]
This should be fixed by #46 (which will be in an upcoming release). Please let me know if not and we can reopen.
